hail/python/hail/expr/type_parsing.py

```diff
@@ -4,7 +4,8 @@
 
 type_grammar = Grammar(
     r"""
-    type = _ (array / set / dict / struct / tuple / interval / int64 / int32 / float32 / float64 / bool / str / call / str / locus) _
+    type = _ (array / set / dict / struct / tuple / interval / int64 / int32 / float32 / float64 / bool / str / call / str / locus / void) _
```

Comment:
I added void to the parsing grammar here, but not the documentation for `dtype` since the void type shouldn't be user visible.